### PR TITLE
added xml_file_path argument to neuroscope sorting; added docstring; …

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
Attn: @h-mayorquin 

Changes:
1. Added `xml_file_path` argument to NeuroscopeSortingInterface (depends on https://github.com/SpikeInterface/spikeextractors/pull/664)
2. Added docstring for NeuroscopeSortingInterface
3. Disabled waveform setting since, for NWB conversion purposes it's actually not working right now anyway. Once we get that upstream stuff done we can actually add a proper feature for it

## Motivation

Bug fix for when multiple `.xml` files exist in the same folder for NeuroscopeSorting data.

## How to test the behavior?
It's been on the to-do list for the NEO folks to add some NeuroscopeSorting data; I'll check in with them. Otherwise it's a bit hard to test this explicitly until we add that to GIN coverage.

## Checklist

- [X] Have you checked our [Contributing](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/docs/contribute.rst) document?
- [X] Have you ensured the PR description clearly describes the problem and solutions?
- [X] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/nwb-conversion-tools/pulls) for the same change?

### CODE REVIEW

**Review by h-mayorquin — APPROVED** [review]
I used this and it solve the previous problem that I had.

File: nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py [hunk 1]
```diff
@@ -226,24 +226,67 @@ def __init__(
...
+            # load_waveforms=load_waveforms,
+            # gain=gain,
         )
 
     def get_metadata(self):
         session_path = Path(self.source_data["folder_path"])
         session_id = session_path.stem
-        metadata = NeuroscopeRecordingInterface.get_ecephys_metadata(
+        metadata = dict()
+        metadata["Ecephys"] = NeuroscopeRecordingInterface.get_ecephys_metadata(
```
Comment (bendichter) [inline nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py:288]: shouldn't we be using the defined `xml_file_path` if it is provided?

Comment (bendichter) [reply nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py:288]: or is that done somewhere else?

Comment (CodyCBakerPhD) [reply nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py:288]: Oh, good catch! Fixing now.

File: nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py [hunk 2]
```diff
@@ -226,24 +228,60 @@ def __init__(
...
+        """
         assert HAVE_LXML, INSTALL_MESSAGE
 
         super().__init__(
             folder_path=folder_path,
             keep_mua_units=keep_mua_units,
             exclude_shanks=exclude_shanks,
-            load_waveforms=load_waveforms,
-            gain=gain,
+            xml_file_path=xml_file_path,
```
Comment (h-mayorquin) [inline nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py:274]: This generates an error as it stands: the class from where this inherited does not expect a `xml_file_path`.

    ---------------------------------------------------------------------------
    TypeError                                 Traceback (most recent call last)
    /tmp/ipykernel_11701/3205164773.py in <module>
         78         print(clu_matches_in_session[0].name, clu_matches_in_session[0].suffix)
         79         if '1' in clu_matches_in_session[0].suffix:
    ---> 80             multisorting_neuroscope = NeuroscopeSortingInterface(folder_path=session_path, keep_mua_units=False, xml_file_path=str(xml_file_path))
         81             source_data.update(
         82                 NeuroscopeSorting=dict(
    
    ~/development/nwb-conversion-tools/nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py in __init__(self, folder_path, keep_mua_units, exclude_shanks, xml_file_path)
        272             keep_mua_units=keep_mua_units,
        273             exclude_shanks=exclude_shanks,
    --> 274             xml_file_path=xml_file_path,
        275             # TODO: we can enable this once
        276             #     a) waveforms on unit columns support conversion factor in NWB
    
    ~/development/nwb-conversion-tools/nwb_conversion_tools/datainterfaces/ecephys/basesortingextractorinterface.py in __init__(self, **source_data)
         26     def __init__(self, **source_data):
         27         super().__init__(**source_data)
    ---> 28         self.sorting_extractor = self.SX(**source_data)
         29         self.writer_class = map_si_object_to_writer(self.sorting_extractor)(self.sorting_extractor)
         30 
    
    TypeError: __init__() got an unexpected keyword argument 'xml_file_path'

**Review by h-mayorquin — COMMENTED** [review]
After the last changes were made I have an error again. Commenting the mentioned line (the xml_file_path argument in the `super().__init__` makes the script work.

**Comment by h-mayorquin** [comment]
Also, now when when I am using the latest fetch of `nwb_conversion_tools` from master, this still does not work for for double xml sessions. Self-contained example:

```python
    from pathlib import Path
    from nwb_conversion_tools import NeuroscopeSortingInterface
    
    double_xml_sessions_path = [
        Path("/shared/catalystneuro/Buzsaki/TingleyD/DT5/20161021_1840um_2412um_merge"),
        Path("/shared/catalystneuro/Buzsaki/TingleyD/DT5/20161025_1944um_2232um_merge"),
        Path("/shared/catalystneuro/Buzsaki/TingleyD/DT7/20170317_288um_0um_merge"),
    ]
    session_path = double_xml_sessions_path[0]  # Chose any, it does not work for them.
    session_id = session_path.name
    xml_file_path = session_path / f"{session_id}.xml"
    print(xml_file_path)
    print(xml_file_path.is_file())
    multisorting_neuroscope = NeuroscopeSortingInterface(folder_path=str(session_path), keep_mua_units=False, xml_file_path=str(xml_file_path))
```

This produces the following error trace.

```python
    ---------------------------------------------------------------------------
    AssertionError                            Traceback (most recent call last)
    /tmp/ipykernel_15928/2519331352.py in <module>
         12 print(xml_file_path)
         13 print(xml_file_path.is_file())
    ---> 14 multisorting_neuroscope = NeuroscopeSortingInterface(folder_path=str(session_path), keep_mua_units=False, xml_file_path=str(xml_file_path))
    
    ~/development/nwb-conversion-tools/nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py in __init__(self, folder_path, keep_mua_units, exclude_shanks, xml_file_path)
        271             folder_path=folder_path,
        272             keep_mua_units=keep_mua_units,
    --> 273             exclude_shanks=exclude_shanks,
        274             #xml_file_path=xml_file_path,
        275             # TODO: we can enable this once
    
    ~/development/nwb-conversion-tools/nwb_conversion_tools/datainterfaces/ecephys/basesortingextractorinterface.py in __init__(self, **source_data)
         26     def __init__(self, **source_data):
         27         super().__init__(**source_data)
    ---> 28         self.sorting_extractor = self.SX(**source_data)
         29         self.writer_class = map_si_object_to_writer(self.sorting_extractor)(self.sorting_extractor)
         30 
    
    ~/development/spikeextractors/spikeextractors/extractors/neuroscopeextractors/neuroscopeextractors.py in __init__(self, folder_path, keep_mua_units, exclude_shanks, load_waveforms, gain)
        574         xml_files = [f for f in folder_path.iterdir() if f.is_file if f.suffix == ".xml"]
        575         assert len(xml_files) > 0, "No .xml file found in the folder!"
    --> 576         assert len(xml_files) == 1, "More than one .xml file found in the folder!"
        577         xml_filepath = xml_files[0]
        578 
    
    AssertionError: More than one .xml file found in the folder!

File: nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py (hunk 2 above)
Comment (CodyCBakerPhD) [reply nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py:274]: @h-mayorquin You will have to make sure you are also using `master` of `spikeextractors` until they do a new release, see https://github.com/SpikeInterface/spikeextractors/blob/master/spikeextractors/extractors/neuroscopeextractors/neuroscopeextractors.py#L576

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin Please raise re-occurrences of previous issues outside of a merged PR; these are much less likely to be noticed as they are at the bottom of the stack.

This part of the second error trace is the issue

tools/nwb_conversion_tools/datainterfaces/ecephys/neuroscope/neuroscopedatainterface.py in __init__(self, folder_path, keep_mua_units, exclude_shanks, xml_file_path)
        271             folder_path=folder_path,
        272             keep_mua_units=keep_mua_units,
    --> 273             exclude_shanks=exclude_shanks,
        274             #xml_file_path=xml_file_path,

because of the comment in front of `xml_file_path`, which I don't see on `master` since I think you added it manually folllowing the previous error trace. Basically, this issue is entirely expected because it means the `xml_file_path` isn't being passed to the extractor, which is what was happening before the fix.

**Pull request merged**